library/spdm_requester_lib/libspdm_req_communication.c

```diff
@@ -53,6 +55,7 @@ libspdm_return_t libspdm_start_session(void *spdm_context, bool use_psk,
 
     if (!use_psk) {
     #if LIBSPDM_ENABLE_CAPABILITY_KEY_EX_CAP
+        LIBSPDM_ASSERT((psk_hint == NULL) && (psk_hint_size == 0));
```

Comment:
These parameters can just be ignored if it's not a PSK session. No need to force Integrator to set their values.

Reply:
why?
There is no meaning to input PSK_HINT for DHE session. I would treat it as user mistake.

Reply:
No.  libspdm does not evaluate psk_hint or psk_hint_size if the session is not PSK.

My intention is:
1. For DHE session, PSK_HINT must be NULL.
2. For PSK session, PSK_HINT could be NULL or non-NULL.

I am not sure why we need allow DHE with PSK_HINT non-NULL, and ignore them.
What is the use case we want to support?

Reply:
>What is the use case we want to support?

No use case. Just in the documentation I would like it to say
> If `use_psk` is false then the `psk_hint` and `psk_hint_size` parameters are ignored.

Reply:
updated.